exporter/elasticsearchexporter/model.go

```diff
@@ -110,7 +110,8 @@ func (m *encodeModel) encodeLog(resource pcommon.Resource, resourceSchemaURL str
 	default:
 		document = m.encodeLogDefaultMode(resource, record, scope)
 	}
-	document.Dedup()
+	// For OTel mode, prefix conflicts are not a problem as otel-data has subobjects: false
+	document.Dedup(m.mode != MappingOTel)
```

Comment:
`Dedup` is really doing two things, ensuring attributes can be dot-expanded (by appending `.value` if there's a prefix conflict), and making sure that there are no duplicate attributes.

Don't we want to still dedup attributes in OTel mode?

Reply:
> Dedup is really doing two things, ensuring attributes can be dot-expanded (by appending .value if there's a prefix conflict), and making sure that there are no duplicate attributes.

document.Dedup(true) will do both, while document.Dedup(false) will only do the latter. (I reckon this is poor API design)

Reply:
oh, right